src/components/country-challenges-chart/country-challenges-chart-selectors.js

```diff
@@ -48,7 +63,36 @@ const getFilteredData = createSelector(
     const relatedCountries = selectedCountryRelations[selectedFilter];
     return plotRawData.filter(country => relatedCountries.includes(country.iso));
   }
-)
+);
+
+const getFilteredAvailableData = createSelector(
+  [getFilteredData, getCountryChallengesSelectedKey],
+  (data, selectedKey) => {
+    if (!data || !selectedKey) return null;
+    return data.filter(
+      (d) => d.xAxisValues[selectedKey] || d.xAxisValues[selectedKey] === 0
+    );
+  }
+);
```

Comment:
You don't need the computation inside this selector. If you use the `getFilteredData` as source for the `data` prop the buggy animation does not happens anymore. 
```suggestion

```